Octokit/Clients/RepositoryHooksClient.cs

```diff
@@ -0,0 +1,115 @@
+        /// <param name="hookId"></param>
+        /// <returns></returns>
+        /// <remarks>See <a href="http://developer.github.com/v3/repos/hooks/#get-single-hook">API documentation</a> for more information.</remarks>
+        public Task<RepositoryHook> Get(string owner, string repositoryName, int hookId)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(owner, "owner");
+            Ensure.ArgumentNotNullOrEmptyString(repositoryName, "repositoryName");
+
+            return ApiConnection.Get<RepositoryHook>(ApiUrls.RepositoryHookById(owner, repositoryName, hookId));
+        }
+
+        /// <summary>
+        /// Creates a hook for a repository
+        /// </summary>
+        /// <remarks>See <a href="http://developer.github.com/v3/repos/hooks/#create-a-hook">API documentation</a> for more information.</remarks>
+        /// <returns></returns>
+        public Task<RepositoryHook> Create(string owner, string repositoryName, NewRepositoryHook hook)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(owner, "owner");
+            Ensure.ArgumentNotNullOrEmptyString(repositoryName, "repositoryName");
+            Ensure.ArgumentNotNull(hook, "hook");
+
+            return ApiConnection.Post<RepositoryHook>(ApiUrls.RepositoryHooks(owner, repositoryName), hook);
+        }
+
+        /// <summary>
+        /// Edits a hook for a repository
+        /// </summary>
+        /// <remarks>See <a href="http://developer.github.com/v3/repos/hooks/#edit-a-hook">API documentation</a> for more information.</remarks>
+        /// <returns></returns>
+        public Task<RepositoryHook> Edit(string owner, string repositoryName, int hookId, EditRepositoryHook hook)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(owner, "owner");
+            Ensure.ArgumentNotNullOrEmptyString(repositoryName, "repositoryName");
+            Ensure.ArgumentNotNull(hook, "hook");
+
+            return ApiConnection.Patch<RepositoryHook>(ApiUrls.RepositoryHookById(owner, repositoryName, hookId), hook);
+        }
+
+        /// <summary>
+        /// Tests a hook for a repository
+        /// </summary>
+        /// <remarks>See <a href="http://developer.github.com/v3/repos/hooks/#test-a-hook">API documentation</a> for more information. 
+        /// This will trigger the hook with the latest push to the current repository if the hook is subscribed to push events. If the hook 
+        /// is not subscribed to push events, the server will respond with 204 but no test POST will be generated.</remarks>
+        /// <returns></returns>
+        public Task Test(string owner, string repositoryName, int hookId)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(owner, "owner");
+            Ensure.ArgumentNotNullOrEmptyString(repositoryName, "repositoryName");
+
+            return ApiConnection.Post<RepositoryHookTestRequest>(ApiUrls.RepositoryHookTest(owner, repositoryName, hookId), new RepositoryHookTestRequest());
```

Comment:
Ok, now that I understand the reason for the empty classes let me see if I can suggest an alternative:

Given both these methods should return `204 No Content`, should we be inspecting the response code? With many of our Delete actions we use `Task<HttpStatusCode>` to check that nothing unexpected happened (see `WatchedClient.UnwatchRepo` for an example).

We don't have an equivalent overload for `Post` right now, but given you have two scenarios here which seem to fit it I think it's worth incorporating.

Thoughts?